src/main/java/net/dv8tion/jda/core/entities/impl/TextChannelImpl.java

```diff
@@ -223,6 +223,11 @@ public String getTopic()
     }
 
     @Override
+    public boolean isNSFW() {
+        return name.equals("nsfw") || name.startsWith("nsfw-");
```

Comment:
Change to ``return name.equalsIgnoreCase("nsfw") || name.toLowerCase().startsWith("nsfw-");`` should fix it